addons/website_sale/static/src/snippets/s_products_searchbar/options.js

```diff
@@ -25,41 +22,32 @@ options.registry.ProductsSearchBar = options.Class.extend({
     /**
      * @see this.selectClass for parameters
      */
-    openSearchbarSettings: function (previewMode, widgetValue, params) {
-        var self = this;
-        new Dialog(this, {
-            title: _t("Products Search Bar"),
-            $content: $(qweb.render('website_sale.dialog.productsSearchBar', {
-                currentOrderBy: this.$searchOrderField.val(),
-                currentLimit: parseInt(this.$searchProductsInput.attr('data-limit')),
-                currentDisplayDescription: this.$searchProductsInput.attr('data-display-description') === 'true',
-                currentDisplayPrice: this.$searchProductsInput.attr('data-display-price') === 'true',
-                currentDisplayImage: this.$searchProductsInput.attr('data-display-image') === 'true',
-            })),
-            buttons: [
-                {
-                    text: _t("Save"),
-                    classes: 'btn-primary',
-                    click: function () {
-                        self.$searchOrderField.attr({
-                            'value': this.$('#order_by').val(),
-                        });
-                        self.$searchProductsInput.attr({
-                            'data-limit': this.$('#use_autocomplete').is(':checked') ? this.$('#limit').val() : 0,
-                            'data-display-description': this.$('#display_description').is(':checked'),
-                            'data-display-price': this.$('#display_price').is(':checked'),
-                            'data-display-image': this.$('#display_image').is(':checked'),
-                        });
-                        self.$target.trigger('content_changed');
-                        this.close();
-                    },
-                },
-                {
-                    text: _t("Discard"),
-                    close: true,
-                },
-            ],
-        }).open();
+    useAutocomplete(previewMode, widgetValue, params) {
+        this.$target.attr({ 'data-limit': widgetValue === '' ? 0 : 5 });
+    },
+    /**
+     * @see this.selectClass for parameters
+     */
+    setInputLimit(previewMode, widgetValue, params) {
+        const value = Math.min(Math.max(parseInt(widgetValue), 0), 20);
+        this.$target.attr({ 'data-limit': value || 0 });
+    },
```

Comment:
I think we can forget about the 0, 20 limit. If users put less or more, it is ok. Then you can remove the `setInputLimit`option and use the standard `data-select-data-attribute`